File: lib/cs/MsgPack/Internal/Format.cs
namespace ILib.MsgPack
{
	public static class Format
	{
		public const byte PositiveFixIntMin = 0x00;
		public const byte PositiveFixIntMax = 0x7f;
		public const byte FixMapMin = 0x80;
		public const byte FixMapMax = 0x8f;
		public const byte FixArrayMin = 0x90;
		public const byte FixArrayMax = 0x9f;
		public const byte FixStrMin = 0xa0;
		public const byte FixStrMax = 0xbf;
		public const byte Nil = 0xc0;
		public const byte NeverUsed = 0xc1;
		public const byte False = 0xc2;
		public const byte True = 0xc3;
		public const byte Bin8 = 0xc4;
		public const byte Bin16 = 0xc5;
		public const byte Bin32 = 0xc6;
		public const byte Ext8 = 0xc7;
		public const byte Ext16 = 0xc8;
		public const byte Ext32 = 0xc9;
		public const byte Float32 = 0xca;
		public const byte Float64 = 0xcb;
		public const byte UInt8 = 0xcc;
		public const byte UInt16 = 0xcd;
		public const byte UInt32 = 0xce;
		public const byte UInt64 = 0xcf;
		public const byte Int8 = 0xd0;
		public const byte Int16 = 0xd1;
		public const byte Int32 = 0xd2;
		public const byte Int64 = 0xd3;
		public const byte FixExt1 = 0xd4;
		public const byte FixExt2 = 0xd5;
		public const byte FixExt4 = 0xd6;
		public const byte FixExt8 = 0xd7;
		public const byte FixExt16 = 0xd8;
		public const byte Str8 = 0xd9;
		public const byte Str16 = 0xda;
		public const byte Str32 = 0xdb;
		public const byte Array16 = 0xdc;
		public const byte Array32 = 0xdd;
		public const byte Map16 = 0xde;
		public const byte Map32 = 0xdf;
		public const byte NegativeFixIntMin = 0xe0;
		public const byte NegativeFixIntMax = 0xff;

		public static bool IsPositiveFixInt(byte format)
		{
			return Format.PositiveFixIntMax >= format && format >= Format.PositiveFixIntMin;
		}

		public static bool IsNegativeFixInt(byte format)
		{
			return Format.NegativeFixIntMax >= format && format >= Format.NegativeFixIntMin;
		}

		public static bool IsFixMap(byte format)
		{
			return Format.FixMapMax >= format && format >= Format.FixMapMin;
		}

		public static bool IsFixArray(byte format)
		{
			return Format.FixArrayMax >= format && format >= Format.FixArrayMin;
		}

		public static bool IsFixStr(byte format)
		{
			return Format.FixStrMax >= format && format >= Format.FixStrMin;
		}

		public static bool IsArrayFamily(byte format)
		{
			return IsFixArray(format) || Array16 == format || Array32 == format;
		}

		public static bool IsMapFamily(byte format)
		{
			return IsFixMap(format) || Map16 == format || Map32 == format;
		}

	}
}

File: lib/golang/format.go
package protopack

const (
	formatPositiveFixIntMin byte = 0x00
	formatPositiveFixIntMax byte = 0x7f
	formatFixMapMin         byte = 0x80
	formatFixMapMax         byte = 0x8f
	formatFixArrayMin       byte = 0x90
	formatFixArrayMax       byte = 0x9f
	formatFixStrMin         byte = 0xa0
	formatFixStrMax         byte = 0xbf
	formatNil               byte = 0xc0
	formatNeverUsed         byte = 0xc1
	formatFalse             byte = 0xc2
	formatTrue              byte = 0xc3
	formatBin8              byte = 0xc4
	formatBin16             byte = 0xc5
	formatBin32             byte = 0xc6
	formatExt8              byte = 0xc7
	formatExt16             byte = 0xc8
	formatExt32             byte = 0xc9
	formatFloat32           byte = 0xca
	formatFloat64           byte = 0xcb
	formatUInt8             byte = 0xcc
	formatUInt16            byte = 0xcd
	formatUInt32            byte = 0xce
	formatUInt64            byte = 0xcf
	formatInt8              byte = 0xd0
	formatInt16             byte = 0xd1
	formatInt32             byte = 0xd2
	formatInt64             byte = 0xd3
	formatFixExt1           byte = 0xd4
	formatFixExt2           byte = 0xd5
	formatFixExt4           byte = 0xd6
	formatFixExt8           byte = 0xd7
	formatFixExt16          byte = 0xd8
	formatStr8              byte = 0xd9
	formatStr16             byte = 0xda
	formatStr32             byte = 0xdb
	formatArray16           byte = 0xdc
	formatArray32           byte = 0xdd
	formatMap16             byte = 0xde
	formatMap32             byte = 0xdf
	formatNegativeFixIntMin byte = 0xe0
	formatNegativeFixIntMax byte = 0xff
)

func isPositiveFixInt(format byte) bool {
	return formatPositiveFixIntMax >= format && format >= formatPositiveFixIntMin
}

func isNegativeFixInt(format byte) bool {
	return formatNegativeFixIntMax >= format && format >= formatNegativeFixIntMin
}

func isFixMap(format byte) bool {
	return formatFixMapMax >= format && format >= formatFixMapMin
}

func isFixArray(format byte) bool {
	return formatFixArrayMax >= format && format >= formatFixArrayMin
}

func isFixStr(format byte) bool {
	return formatFixStrMax >= format && format >= formatFixStrMin
}

func isArrayFamily(format byte) bool {
	return isFixArray(format) || formatArray16 == format || formatArray32 == format
}

func isMapFamily(format byte) bool {
	return isFixMap(format) || formatMap16 == format || formatMap32 == format
}
